schema/integer_test.go

```diff
@@ -28,6 +29,18 @@ func TestIntegerValidator(t *testing.T) {
 	s, err = Integer{Boundaries: &Boundaries{Min: 2, Max: 10}}.Validate(3)
 	assert.NoError(t, err)
 	assert.Equal(t, 3, s)
+	s, err = Integer{Boundaries: &Boundaries{Min: math.Inf(-1), Max: 10}}.Validate(3)
+	assert.NoError(t, err)
+	assert.Equal(t, 3, s)
+	s, err = Integer{Boundaries: &Boundaries{Min: math.NaN(), Max: 10}}.Validate(3)
+	assert.NoError(t, err)
+	assert.Equal(t, 3, s)
+	s, err = Integer{Boundaries: &Boundaries{Min: 2, Max: math.Inf(1)}}.Validate(3)
+	assert.NoError(t, err)
+	assert.Equal(t, 3, s)
+	s, err = Integer{Boundaries: &Boundaries{Min: 2, Max: math.NaN()}}.Validate(3)
+	assert.NoError(t, err)
+	assert.Equal(t, 3, s)
```

Comment:
Same here. Not sure if support for NaN and Inf values was intended to work, but since it does work, I added some tests for it.